torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -373,11 +375,16 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
   pipeWrite(
       clientPipe.pipe_,
       std::move(requestMessage),
-      [this, &clientPipe, futureResponseMessage](
-          const tensorpipe::Error& error) {
+      [this, &clientPipe, requestMessageId](const tensorpipe::Error& error) {
```

Comment:
Is there no other way to avoid the shadowing? By storing just the ID we must do a lookup which, although fast, wouldn't be needed if we already had the pointer. (As this is only relevant in the error codepath it's not really important... mostly a matter of style)

Reply:
We can simply change the variable name, although I decided to take this approach to also get rid of the additional copy into the lambda.